ergo/platforms/predictit.py

```diff
@@ -0,0 +1,332 @@
+        if data['timeStamp'] == 'N/A':
+            self.timestamp = None
+        else:
+            self.timestamp = parse(data['timeStamp'])
+
+    @property
+    def questions(self) -> Iterator[PredictItQuestion]:
+        """
+        Generate all of the questions in the market.
+        :return: iterator of questions in market
+        """
+        for data in self._data['contracts']:
+            yield PredictItQuestion(self, data)
+
+    def refresh(self):
+        """
+        Refetch the market data from PredictIt,
+        used when the question data might have changed.
+        """
+        r = self.predictit._get(self.api_url)
+        self._load_attr(r.json())
+
+    def get_question_by_id(self, id: int) -> PredictItQuestion:
+        """
+        Return the specified question given by the id number.
+        :param id:
+        :return: question
+        """
+        for question in self.questions:
+            if question.id == id:
+                return question
+        raise ValueError("Unable to find a question with that id.")
+
+    def search_question(self, name: str) -> PredictItQuestion:
+        """
+        Return the specified question given by the name of the question,
+        using fuzzy matching in the case where the name isn't exact.
+        :param name:
+        :return: question
+        """
+        guess = re.sub(r'[^\w\s]', '', name).lower()
+        guess_words = guess.split()
+        most_matches = 0
+        best_diff = 0
+        best_diff_contract = None
+        for contract in self.questions:
+            short_name = re.sub(r'[^\w\s]', '', contract.name).lower()
+            matches = sum([word in short_name for word in guess_words])
+            diff = fuzz.token_sort_ratio(guess, short_name)
+            if matches > most_matches or (matches >= most_matches and (diff > best_diff)):
+                best_diff = diff
+                best_diff_contract = contract
+            if matches > most_matches:
+                most_matches = matches
+        if best_diff_contract is None:
+            raise ValueError("Unable to find a question with that name.")
+        return best_diff_contract
+
+
+class PredictIt:
+    """
+    The main class for interacting with PredictIt.
+    """
+
+    def __init__(self):
+        self.api_url = "https://www.predictit.org/api/marketdata"
+        self.s = requests.Session()
+        self._data = self._get(f"{self.api_url}/all/").json()
+
+    def _get(self, url):
+        """
+        Send a get request to to PredictIt API.
+        :param url:
+        :return: response
+        """
+        r = self.s.get(url)
+        assert "Slow down!" not in str(r.content), "Hit API rate limit"
+        return r
+
+    def refresh_markets(self):
+        """
+        Refetch all of the markets from the predictit api.
+        """
+        self._data = self._get(f"{self.api_url}/all/").json()
+
+    def get_markets(self) -> Iterator[PredictItMarket]:
```

Comment:
Why `.get_markets()` here but `.questions` above? Pick one for consistency (probably property notation)?